cmd/soroban-cli/Cargo.toml

```diff
@@ -126,15 +123,18 @@ glob = "0.3.1"
 open = "5.3.0"
 url = "2.5.2"
 
-# For hyper-tls
 [target.'cfg(unix)'.dependencies]
-openssl = { version = "=0.10.55", features = ["vendored"] }
+reqwest = { version = "0.12.7", features = [
+    "json",
+    "blocking",
+    "stream",
+    "native-tls-vendored",
```

Comment:
We should check which tls the https://github.com/stellar/rs-stellar-rpc-client uses, so that both ideally use the same. I thought it used rustls, but I'm not sure.

Reply:
```
stellar-rpc-client v21.4.0
├── jsonrpsee-http-client v0.20.3
│   ├── hyper-rustls v0.24.2
...
```

I think it uses hyper-rustls. Do you want to use hyper-rustls on all platforms? Reqwest defaults to using the system's TLS library. See https://github.com/seanmonstar/reqwest?tab=readme-ov-file#requirements

Reply:
We switched to rustls-tls. e128ae956089bc6a6aafe9db589902da0c5bc11b

https://github.com/seanmonstar/reqwest/issues/2025
https://github.com/rust-lang/rustup/issues/3400#issuecomment-1683762630